frontend/src/FullPageRoutes/RecordPage.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { useReactMediaRecorder } from 'react-media-recorder';
+import { Grid, makeStyles } from '@material-ui/core';
+import Controls from '../components/video/Controls';
+import { request } from '../utils/fetch';
+import useFocusedUser from '../utils/context';
+
+const useStyles = makeStyles({
+    video_root: {
+        backgroundColor: 'black',
+        width: 1066,
+        height: 700,
+    },
+});
+
+type PreviewProps = {
+    stream: MediaStream | null;
+};
+function Preview({ stream }: PreviewProps) {
+    const ref = React.useRef<HTMLVideoElement>(null);
+    const classes = useStyles();
+
+    React.useEffect(() => {
+        if (ref.current && stream) {
+            ref.current.srcObject = stream;
+        }
+    }, [stream]);
+
+    if (!stream) {
+        return null;
+    }
+    return <video ref={ref} className={classes.video_root} autoPlay />;
+}
+
+function sendVideo(blob: Blob, receiver: number) {
+    const reader = new FileReader();
+    reader.readAsDataURL(blob);
+    reader.onloadend = () => {

```

Comment:
Prefer [`reader.load`](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/load_event) over [`reader.loadend`](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/loadend_event) here because loadend will be called even if the file read fails whereas load only gets called after a successful file read.